test/Microsoft.DotNet.Tools.Tests.Utilities/TestBase.cs

```diff
@@ -178,5 +193,32 @@ private bool IsPortable(string executablePath)
             Console.WriteLine(runtimeConfig.Framework);
             return runtimeConfig.IsPortable;
         }
+
+        protected ProjDir NewDir([CallerMemberName] string callingMethod = nameof(NewDir), string identifier = "")
+        {
+            return new ProjDir(
+                TestAssetsManager.CreateTestDirectory(callingMethod: callingMethod, identifier: identifier).Path);
+        }
+
+        protected ProjDir NewLib([CallerMemberName] string callingMethod = nameof(NewDir), string identifier = "")
+        {
+            var dir = NewDir(callingMethod: callingMethod, identifier: identifier);
+
+            try
+            {
+                new NewCommand()
+                    .WithWorkingDirectory(dir.Path)
+                    .ExecuteWithCapturedOutput("-t Lib")
+                .Should().Pass();
+            }
+            catch (System.ComponentModel.Win32Exception e)
+            {
+                throw new Exception(
+                    $"Intermittent error in `dotnet new` occurred when running it in dir `{dir.Path}`\nException:\n{e}");
```

Comment:
that is some combination of dotnet new and test assets manager doing some stuff at the same time